### PR TITLE
Updated third party file

### PR DESCRIPTION
**Pull request by asifsmohammed**
Signed-off-by: Asif Sohail Mohammed <nsifmoh@amazon.com>

### Description
Updated THIRD-PARTY dependency file
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: THIRD-PARTY [hunk 1]
```diff
@@ -8211,7 +8211,7 @@ The Apache Software Foundation (http://www.apache.org/).
 
 --------------------------------------------------------------------------------
 
-104. Group: org.apache.logging.log4j  Name: log4j-api  Version: 2.11.1
+104. Group: org.apache.logging.log4j  Name: log4j-api  Version: 2.15.0
```
Comment (sshivanii) [inline THIRD-PARTY:8214]: Can you explain why `log4j-api` version here is 2.15.0 whereas down below on line 8431 it's 2.16.0?

Comment (asifsmohammed) [reply THIRD-PARTY:8214]: This dependency file is auto generated based on the dependencies we use in the Data Prepper.

Comment (sshivanii) [reply THIRD-PARTY:8214]: I know it's auto generated, I'm thinking if there's some place we're using 2.15.0 in `log4j-api` dependency and if we should upgrade that to 2.16.0

Comment (dlvenable) [reply THIRD-PARTY:8214]: Other dependencies may be pulling it in. The authoritative way to determine the actual dependencies at runtime would be to run:

```
./gradlew -p data-prepper-core dependencies
```

**Review by dlvenable — APPROVED** [review]

File: THIRD-PARTY (hunk 1 above)
Comment (dlvenable) [reply THIRD-PARTY:8214]: There may be a future opportunity to pare down some duplicate dependencies in the automation. But, I don't think having both versions in the THIRD-PARTY file is a problem right now.

**Review by sshivanii — APPROVED** [review]

File: THIRD-PARTY (hunk 1 above)
Comment (sshivanii) [reply THIRD-PARTY:8214]: @dlvenable The actual dependencies for `log4j-api` at runtime show all versions as 2.16.0, so I believe we're good and unsure where that 2.15.0 is popping up from.
```
./gradlew -p data-prepper-core dependencies | grep log4j
+--- org.apache.logging.log4j:log4j-bom:2.16.0
|    +--- org.apache.logging.log4j:log4j-core:2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-slf4j-impl:2.16.0 (c)
|    \--- org.apache.logging.log4j:log4j-api:2.16.0 (c)
+--- org.apache.logging.log4j:log4j-core -> 2.16.0
|    \--- org.apache.logging.log4j:log4j-api:2.16.0
+--- org.apache.logging.log4j:log4j-slf4j-impl -> 2.16.0
|    \--- org.apache.logging.log4j:log4j-api:2.16.0
+--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
\--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
+--- org.apache.logging.log4j:log4j-bom:2.16.0 (n)
+--- org.apache.logging.log4j:log4j-core (n)
\--- org.apache.logging.log4j:log4j-slf4j-impl (n)
|    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    |    |    +--- org.apache.logging.log4j:log4j-api:2.11.1 -> 2.16.0
|    |    |    |    +--- org.apache.logging.log4j:log4j-core:2.11.1 -> 2.16.0
|    |    |    |    |    \--- org.apache.logging.log4j:log4j-api:2.16.0
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
+--- org.apache.logging.log4j:log4j-bom:2.16.0
|    +--- org.apache.logging.log4j:log4j-core:2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-slf4j-impl:2.16.0 (c)
|    \--- org.apache.logging.log4j:log4j-api:2.16.0 (c)
+--- org.apache.logging.log4j:log4j-core -> 2.16.0 (*)
+--- org.apache.logging.log4j:log4j-slf4j-impl -> 2.16.0
|    +--- org.apache.logging.log4j:log4j-api:2.16.0
|    \--- org.apache.logging.log4j:log4j-core:2.16.0 (*)
+--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
+--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
+--- org.apache.logging.log4j:log4j-bom:2.16.0
|    +--- org.apache.logging.log4j:log4j-core:2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-slf4j-impl:2.16.0 (c)
|    \--- org.apache.logging.log4j:log4j-api:2.16.0 (c)
+--- org.apache.logging.log4j:log4j-core -> 2.16.0
|    \--- org.apache.logging.log4j:log4j-api:2.16.0
+--- org.apache.logging.log4j:log4j-slf4j-impl -> 2.16.0
|    \--- org.apache.logging.log4j:log4j-api:2.16.0
+--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
\--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    |    |    +--- org.apache.logging.log4j:log4j-api:2.11.1 -> 2.16.0
|    |    |    |    +--- org.apache.logging.log4j:log4j-core:2.11.1 -> 2.16.0
|    |    |    |    |    \--- org.apache.logging.log4j:log4j-api:2.16.0
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
+--- org.apache.logging.log4j:log4j-bom:2.16.0
|    +--- org.apache.logging.log4j:log4j-core:2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-slf4j-impl:2.16.0 (c)
|    \--- org.apache.logging.log4j:log4j-api:2.16.0 (c)
+--- org.apache.logging.log4j:log4j-core -> 2.16.0 (*)
+--- org.apache.logging.log4j:log4j-slf4j-impl -> 2.16.0
|    +--- org.apache.logging.log4j:log4j-api:2.16.0
|    \--- org.apache.logging.log4j:log4j-core:2.16.0 (*)
+--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
+--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)

```

**Pull request merged**